src/app/display-media-request-handler.ts

```diff
@@ -1,19 +1,103 @@
-import { session } from 'electron';
+import { desktopCapturer, ipcMain, session } from 'electron';
+import { NOTIFICATION_WINDOW_TITLE } from '../common/api-interface';
+import { isDevEnv, isMac } from '../common/env';
+import { logger } from '../common/logger';
+import {
+  ICustomBrowserWindowConstructorOpts,
+  windowHandler,
+} from './window-handler';
+import { createComponentWindow, windowExists } from './window-utils';
 
 /**
- * This is currently supported only on macOS 15+.
- * setDisplayMediaRequestHandler injects into navigator.mediaDevices.getDisplayMedia().
- * With the macOS-only option { useSystemPicker: true },
- * everyting is handled natively by the OS.
+ * For MacOS 15+ the { useSystemPicker: true } overrides the code set in the handler,
+ * and uses the native implementation.
  *
- * For all other OSes and versions, the regular screen share flow will be used.
+ * But for other versions and OSes, the code is executed.
  */
 export const setDisplayMediaRequestHandler = () => {
   const { defaultSession } = session;
-
   defaultSession.setDisplayMediaRequestHandler(
-    async (_request, _callback) => {
-      // TODO - Add support for Windows.
+    async (_request, callback) => {
+      logger.info('display-media-request-handler: getting sources');
```

Comment:
I think `isScreenShareEnabled` check should be added here.

Reply:
`isScreenShareEnabled` is configured in Symphony.config in SDA, which allows/disallows screen sharing. Should this configuration really live in SDA? To me this is controlled via the entitlements in SFE-RTC? @swasunb what do you think?

My suggestion is that we add a check for system permission, but this extra media configuration could perhaps be deprecated here.

Reply:
Less reduce multiple sources of truth if we can

Reply:
what happens when there is no "Control access to screen" permission given to SDA? 
Do we have a posibility to ask it when you click on screenshare button? or nothing happens then?

Reply:
As I understand it, on Windows there's no concept of screen share permission so it'll always be allowed. See this documentation - https://www.electronjs.org/docs/latest/api/system-preferences#systempreferencesgetmediaaccessstatusmediatype-windows-macos

Using the new `useSystemPicker` there seems to be a bug on Mac where even if I disallow the screen sharing still works.

Therefore, I think we're good to proceed with this.